Dockerfile

```
@@ -443,29 +443,73 @@ RUN \
 
 FROM sdk-libc as sdk-go-prep
 
-ENV GOVER="1.21.9"
+# Set up the environment for building.
+ENV GOOS="linux"
+ENV CGO_ENABLED=1
+ENV CFLAGS="-O2 -g -pipe -Wformat -Werror=format-security -Wp,-D_FORTIFY_SOURCE=2 -Wp,-D_GLIBCXX_ASSERTIONS -fexceptions -fstack-clash-protection"
+ENV CXXFLAGS="${CFLAGS}"
+ENV LDFLAGS="-Wl,-z,relro -Wl,-z,now"
+ENV CGO_CFLAGS="${CFLAGS}"
+ENV CGO_CXXFLAGS="${CXXFLAGS}"
+ENV CGO_LDFLAGS="${LDFLAGS}"
+
+ENV GO111MODULE="auto"
+
 ENV AWS_LC_FIPS_VER="2.0.9"
 
 USER root
 RUN dnf -y install golang
 
+ENV GOVER_1_21="1.21.9"
+ENV GOVER_1_22="1.22.1"
```

nit: this is more in line with the convention:
```suggestion
ENV GO121VER="1.21.9"
ENV GO122VER="1.22.1"
```